cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -2055,6 +2052,41 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 maybe_input_insn_via_conv(ctx, inputs[0], Opcode::Bconst, Opcode::Breduce)
             {
                 lower_splat_const(ctx, rd, ctx.get_constant(insn).unwrap(), size);
+            } else if let Some((_, insn)) = maybe_input_insn_multi(
+                ctx,
+                inputs[0],
+                &[
+                    Opcode::Uload8,
+                    Opcode::Sload8,
+                    Opcode::Uload16,
+                    Opcode::Sload16,
+                    Opcode::Uload32,
+                    Opcode::Sload32,
+                    Opcode::Load,
+                ],
+            ) {
+                ctx.sink_inst(insn);
```

Comment:
I just had a quick look at #2366, so I might be missing something, but the description says:
`This can only be done to an instruction with no uses of its result register(s), because it will cause the instruction not to be codegen'd at its original location.`
(I guess the meaning is that there are no uses of its result except the instruction it has been merged into)
It is not clear to me how we ensure that here; at the very least shouldn't there be a test case where the result of a load is used both by a splat and by another operation?

Reply:
Indeed, no other uses than the one by the input merging the op (I'll clarify that, and add a test for the multi-use case).

The condition is ensured by the logic in `maybe_get_input_as_source_or_const()`: it will return `None` as the `inst` field in the returned `LowerInput` if any of the conditions for code motion are not satisfied. So, things just transparently work in the pattern-matching: we won't see any source op at all if we aren't allowed to merge it.